Source/OnlineSubsystemEpic/Private/OnlineSubsystemEpicTypes.h

```diff
@@ -4,37 +4,252 @@
 #include "OnlineSubsystemEpicPackage.h"
 #include "OnlineSubsystemTypes.h"
 #include "IPAddress.h"
+#include "eos_sdk.h"
 
+#define LOGIN_TYPE_EAS TEXT("EAS")
+#define LOGIN_TYPE_CONNECT TEXT("CONNECT")
+
+/**
+ * Wraps the native product user id (PUID) and epic account id (EAID).
+ *
+ * Since the identity interface primarily uses the connect interface
+ * to retrieve user data, the internal PUID will always be valid and
+ * thus is used as basis for this unique net id. Should the user opt
+ * to use the epic accout system login flow, the EAID will be valid
+ * in addition to the PUID.
+ * In general a user should never have to access the PUID or EAID
+ * themselves, if this should become necessary at some point, both id's
+ * can be accessed via FUserOnlineAccount::GetAuthAttribute() with
+ * the following keys:
+ * - PUID: AUTH_ATTR_ID_TOKEN
+ * - EAID: AUTH_ATTR_EA_TOKEN
+ */
 class FUniqueNetIdEpic
-	: public FUniqueNetIdString
+	: public FUniqueNetId
 {
+	// Used purely for GetType()
+	FName Type = EPIC_SUBSYSTEM;
+
+	// Always valid
+	EOS_ProductUserId productUserId;
+
+	// Only valid when using EAS login flow
+	EOS_EpicAccountId epicAccountId;
+
 public:
-	FUniqueNetIdEpic()
-		: FUniqueNetIdString()
+	FUniqueNetIdEpic() = default;
+
+	// Define these to increase visibility to public (from parent's protected)
+	FUniqueNetIdEpic(FUniqueNetIdEpic&&) = default;
+	FUniqueNetIdEpic(const FUniqueNetIdEpic&) = default;
+	FUniqueNetIdEpic& operator=(FUniqueNetIdEpic&&) = default;
+	FUniqueNetIdEpic& operator=(const FUniqueNetIdEpic&) = default;
+
+	virtual ~FUniqueNetIdEpic() = default;
+
+	/** Create a new id from a given string. */
+	explicit FUniqueNetIdEpic(const FString& InUniqueNetId)
```

Comment:
this is the method that the IdentityInterface is using primarily, correct? The interface is passing in the EpicAccountID, are we not storing it because we are expecting other ID's to be given? or did we just miss assigning the InUniqueNetId to the `epicAccountId` property?

Reply:
The identity interface will use the `FUniqueNetIdEpic(EOS_ProductUserId, EOS_EpicAccountId)` constructor. For more see the above conversation